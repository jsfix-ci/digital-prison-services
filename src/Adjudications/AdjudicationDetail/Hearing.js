import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { H3 } from '@govuk-react/heading/lib/presets'
import { LabelAndValue } from './AdjudicationDetail.styles'

export const Hearing = ({ hearing }) => (
  <>
    <H3>Hearing Details</H3>
    {hearing ? (
      <>
        <GridRow>
          <GridCol>
            <LabelAndValue label="Type" value={hearing.hearingType} />
          </GridCol>
          <GridCol>
            <LabelAndValue label="Date" value={hearing.hearingTime} />
          </GridCol>
          <GridCol>
            <LabelAndValue label="Location" value={hearing.location} />
          </GridCol>
          <GridCol>
            <LabelAndValue label="Heard by" value={hearing.heardByName} />
          </GridCol>
          <GridCol>
            <LabelAndValue label="Other representatives" value={hearing.otherRepresentatives} />
          </GridCol>
          <GridCol>
            <LabelAndValue label="Comment" value={hearing.comment} />
          </GridCol>
        </GridRow>
      </>
    ) : (
      <p>No information available</p>
    )}
    <hr />
  </>
)

Hearing.propTypes = {
  hearing: PropTypes.shape({
    hearingType: PropTypes.string,
    hearingTime: PropTypes.string,
    location: PropTypes.string,
    heardByName: PropTypes.string,
    otherRepresentatives: PropTypes.string,
    comment: PropTypes.string,
  }),
}

const mapStateToProps = ({
  adjudicationHistory: {
    detail: { hearing },
  },
}) => ({
  hearing,
})

export default connect(mapStateToProps)(Hearing)
