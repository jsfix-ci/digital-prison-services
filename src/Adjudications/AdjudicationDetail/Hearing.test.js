import React from 'react'
import testRenderer from 'react-test-renderer'

import { Hearing } from './Hearing'

describe('Hearing', () => {
  it('should render missing hearing correctly', () => {
    const wrapper = testRenderer.create(<Hearing />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should render empty data correctly', () => {
    const hearing = {}

    const wrapper = testRenderer.create(<Hearing hearing={hearing} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should render present hearing correctly', () => {
    const hearing = {
      hearingType: 'Adult',
      hearingTime: '12/04/2017 - 12:54',
      location: 'MDI-CR-12',
      heardByName: 'Jane Smith',
      otherRepresentatives: 'CLLR Jones',
      comment: 'Nothing really',
    }

    const wrapper = testRenderer.create(<Hearing hearing={hearing} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
