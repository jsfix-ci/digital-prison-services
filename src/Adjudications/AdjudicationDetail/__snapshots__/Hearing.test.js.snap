// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hearing should render empty data correctly 1`] = `
<div
  className="sc-EHOje jyflZT"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Hearing details
  </h3>
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Type
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        --
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Date
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        --
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Location
      </strong>
      <p
        className="sc-ifAKCX isyHfT"
      >
        --
      </p>
      <p>
        --
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Heard by
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        --
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Other representatives
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        --
      </p>
    </div>
  </div>
</div>
`;

exports[`Hearing should render missing hearing correctly 1`] = `
<div
  className="sc-EHOje jyflZT"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Hearing details
  </h3>
  <p>
    No information available
  </p>
</div>
`;

exports[`Hearing should render present hearing correctly 1`] = `
<div
  className="sc-EHOje jyflZT"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Hearing details
  </h3>
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Type
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        Adult
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Date
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        12/04/2017 - 12:54
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Location
      </strong>
      <p
        className="sc-ifAKCX isyHfT"
      >
        MDI-CR-12
      </p>
      <p>
        --
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Heard by
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        Jane Smith
      </p>
    </div>
    <div
      className="sc-bwzfXH gqOtUI"
    >
      <strong
        className="label"
      >
        Other representatives
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        CLLR Jones
      </p>
    </div>
  </div>
  <div
    className="sc-bdVaJa gkOpjT"
  >
    <div
      className="sc-bwzfXH iXLolC"
    >
      <strong
        className="label"
      >
        Comments
      </strong>
      <p
        className="sc-bxivhb ddqgmM"
      >
        Nothing really
      </p>
    </div>
  </div>
</div>
`;
